lib/ex_doc/autolink.ex

```diff
@@ -0,0 +1,487 @@
+    #
+    # * `:ext` - the extension (`".html"`, "`.xhtml"`, etc)
+    #
+    # * `:skip_undefined_reference_warnings_on` - list of modules to skip the warning on
+    defstruct [
+      :app,
+      :current_module,
+      :module_id,
+      :id,
+      ext: ".html",
+      skip_undefined_reference_warnings_on: []
+    ]
+  end
+
+  alias ExDoc.Formatter.HTML.Templates, as: T
+  alias ExDoc.Refs
+
+  @hexdocs "https://hexdocs.pm/"
+  @otp_docs "http://www.erlang.org/doc/man/"
+
+  def doc(ast, options \\ []) do
+    config = struct!(Config, options)
+
+    ast
+    |> preprocess()
+    |> postprocess(config)
+  end
+
+  defp preprocess(list) when is_list(list) do
+    Enum.map(list, &preprocess/1)
+  end
+
+  defp preprocess(binary) when is_binary(binary) do
+    binary
+  end
+
+  defp preprocess({:pre, _, _} = node) do
+    node
+  end
+
+  defp preprocess({:code, _, [text]} = node) do
+    case parse(text, :regular) do
+      {:ok, link} -> {:a, [href: link], [node]}
+      :error -> node
+    end
+  end
+
+  defp preprocess({:a, attrs, ast} = node) do
+    case custom_link(attrs) do
+      {:ok, link} -> {:a, Keyword.put(attrs, :href, link), ast}
+      :error -> node
+    end
+  end
+
+  defp preprocess({tag, attrs, ast}) do
+    {tag, attrs, preprocess(ast)}
+  end
+
+  defp custom_link(attrs) do
+    with {:ok, href} <- Keyword.fetch(attrs, :href),
+         [[_, text]] <- Regex.scan(~r/^`(.+)`$/, href) do
+      parse(text, :custom_link)
+    else
+      _ -> :error
+    end
+  end
+
+  defp parse("mix help " <> name, _config) do
+    mix_task(name)
+  end
+
+  defp parse("mix " <> name, _config) do
+    mix_task(name)
+  end
+
+  defp parse("t:" <> text, mode) do
+    case parse(text, mode) do
+      {:ok, {:local, :function, name, arity}} ->
+        {:ok, {:local, :type, name, arity}}
+
+      {:ok, {:remote, :function, module, name, arity}} ->
+        {:ok, {:remote, :type, module, name, arity}}
+
+      other ->
+        other
+    end
+  end
+
+  defp parse("c:" <> text, mode) do
+    case parse(text, mode) do
+      {:ok, {:local, :function, name, arity}} ->
+        {:ok, {:local, :callback, name, arity}}
+
+      {:ok, {:remote, :function, module, name, arity}} ->
+        {:ok, {:remote, :callback, module, name, arity}}
+
+      other ->
+        other
+    end
+  end
+
+  defp parse(text, mode) do
+    if String.contains?(text, [" ", "(", ")"]) do
+      :error
+    else
+      case Code.string_to_quoted(text) do
+        {:ok, {:__aliases__, _, _} = module} ->
+          {:ok, {:module, module(module)}}
+
+        {:ok, {:/, _, [{{:., _, [module, name]}, _, []}, arity]}}
+        when is_atom(name) and is_integer(arity) ->
+          {:ok, {:remote, :function, module(module), name, arity}}
+
+        {:ok, {:/, _, [{name, _, nil}, arity]}} when is_atom(name) and is_integer(arity) ->
+          {:ok, {:local, :function, name, arity}}
```

Comment:
@josevalim given https://github.com/erlef/eef-documentation-wg/issues/3#issuecomment-598244066, perhaps it's better to work on the full link here too, i.e.: this would become `{:ok, {config.app, config.current_module, :function, name, arity}}`?

Reply:
never mind, there are some upsides to the current approach, e.g. we neatly convert `{:local, :type, name, arity}` to `{:basic_type, name, arity}` so using this context will make that harder.